File: src/main/java/app/askresume/domain/member/exception/DuplicateMemberException.java
package app.askresume.domain.member.exception;


import app.askresume.global.error.ErrorCode;
import app.askresume.global.error.exception.BusinessException;

public class DuplicateMemberException extends BusinessException {

    public DuplicateMemberException() {
        super(ErrorCode.ALREADY_REGISTERED_MEMBER);
    }
}

File: src/main/kotlin/app/askresume/domain/member/exception/DuplicateMemberException.kt
package app.askresume.domain.member.exception

import app.askresume.global.error.ErrorCodes
import app.askresume.global.error.exception.BusinessException

open class DuplicateMemberException(
    email: String,
    memberType: String,
    override val cause: Throwable? = null,
) : BusinessException(
    codeBook = ErrorCodes.ALREADY_REGISTERED_MEMBER,
    properties = "already.registered.member",
    arguments = arrayOf(email, memberType),
    cause = cause,
)

